### PR TITLE
RethinkDB with Promises

### PR DESCRIPTION
**Pull request by Xstoudi**
Promise functions were missing.

### CODE REVIEW

**Comment by seanhess** [comment]
Sorry I haven't used Typescript since Flow and Elm came out, and I'm too rusty to effectively review this.

File: rethinkdb/rethinkdb.d.ts [hunk 1]
```diff
@@ -34,12 +34,17 @@ declare module "rethinkdb" {
 
 
   export class Cursor {
-    hasNext():boolean;
+    hasNext(): boolean;
     each(cb:(err:Error, row:any)=>void, done?:()=>void);
     each(cb:(err:Error, row:any)=>boolean, done?:()=>void); // returning false stops iteration
+    eachAsync(rowProcess: Function, rowFinal: Function);
```
Comment (RyanCavanaugh) [inline rethinkdb/rethinkdb.d.ts]: Implicit any on this line - please add return type annotation

**Comment by andy-ms** [comment]
This can't be merged because the build is failing.
Prefer using a specific function type (e.g. `(x: number) => string`) over `Function`.
Always use `Promise` as a generic type, even if it's just `Promise<any>`.

**Comment by Xstoudi** [comment]
The whole definitions need to be rewrites actually.
I'll try to find the time to do it.

**Pull request closed without merging**